### PR TITLE
Rearranges in alphabetical order + sort method for Array

### PR DESCRIPTION
**Pull request by jonathan-s**
This PR re-arranges methods in an alphabetical order. It also adds a sort method to array.

### CODE REVIEW

File: src/js.rs [hunk 1]
```diff
@@ -67,133 +67,132 @@ extern {
...
-    ///
-    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
-    #[wasm_bindgen(constructor)]
-    pub fn new() -> Object;
-
-    /// The `hasOwnProperty()` method returns a boolean indicating whether the
-    /// object has the specified property as its own property (as opposed to
-    /// inheriting it).
+    pub type Array;
+    /// The copyWithin() method shallow copies part of an array to another location in the same
```
Comment (fitzgen) [inline src/js.rs]: Nitpick: can you add a newline between `pub type Array` and the doc comment below it?

File: tests/all/js_globals/Array.rs [hunk 2]
```diff
@@ -40,6 +40,38 @@ fn index_of() {
...
+        .file("test.ts", r#"
+            import * as assert from "assert";
+            import * as wasm from "./out";
+
+            export function test() {
+                let numbers = [3, 1, 6, 2];
+                let sorted = wasm.sort_array(numbers);
+
+                for (let [index, number] of sorted.entries()) {
+                    assert.equal([1, 2, 3, 6][index], number);
```
Comment (fitzgen) [inline tests/all/js_globals/Array.rs]: Nitpick: this assertion would be a little simpler with:

```js
assert.deepStrictEqual(sorted, [1, 2, 3, 6])
```

https://nodejs.org/api/assert.html#assert_assert_deepstrictequal_actual_expected_message

**Comment by jonathan-s** [comment]
Changes made :)

**Comment by jonathan-s** [comment]
Looks there is a build error when it tries to download rust.

**Pull request merged**